ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -276,9 +277,27 @@ outputFipLog(std::map<std::string, double>& miscSummaryData,
                               miscSummaryData,
                               regionData);
 
-    if (!substep && !forceDisableFipOutput_) {
+    // For report step 0 we use the RPTSOL config, else derive from RPTSCHED
+    std::unique_ptr<FIPConfig> fipSched;
+    if (reportStepNum != 0) {
+        const auto& rpt = this->schedule_[reportStepNum].rpt_config.get();
+        fipSched = std::make_unique<FIPConfig>(rpt);
+    }
+    const FIPConfig& fipc = reportStepNum == 0 ? this->eclState_.getEclipseConfig().fip()
+                                               : *fipSched;
+
+    if (!substep && !forceDisableFipOutput_ && fipc.output(FIPConfig::OutputField::FIELD)) {
         logOutput_.fip(inplace, this->initialInplace(), "");
-        logOutput_.fip(inplace, this->initialInplace(), "FIPNUM");
+        if (fipc.output(FIPConfig::OutputField::FIPNUM)) {
+            logOutput_.fip(inplace, this->initialInplace(), "FIPNUM");
+        }
+        if (fipc.output(FIPConfig::OutputField::FIP)) {
```

Comment:
This is supposed to handle fip=3

Reply:
> This is supposed to handle fip=3

Right, yeah&ndash;I missed that part.  Sorry about that.